File: leetcode/array/concatenation_of_array.go
package main

import "fmt"

func getConcatenation(nums []int) []int {
	ans := make([]int, len(nums)*2)
	for i := 0; i < len(nums); i++ {
		ans[i] = nums[i]
		ans[i+len(nums)] = nums[i]
	}
	return ans
}

func main() {
	nums := []int{1, 2, 1}
	fmt.Println(getConcatenation(nums))
}

File: leetcode/array/concatenation_of_array.dart
class Solution {
  List<int> getConcatenation(List<int> nums) {
    int size = nums.length;
    final ans = List.filled(size * 2, 0);
    for (var i = 0; i < nums.length; i++) {
      ans[i] = nums[i];
      ans[i + size] = nums[i];
    }
    return ans;
  }
}

void main(List<String> args) {
  final object = Solution();
  final nums = [1, 2, 1];
  print(object.getConcatenation(nums));
}
